File: backend/api/app/Models/Shift.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Shift extends Model
{
    use HasFactory;

    /** Setiap shift dimiliki satu project. */
    public function project()
    {
        return $this->belongsTo(Project::class);
    }

    protected $fillable = [
        'project_id',
        'name',
        'code',
        'start_time',
        'end_time',
        'tolerance_minutes',
        'is_default',
    ];

    public function projects()
    {
        return $this->belongsToMany(Project::class)
            ->withTimestamps()
            ->withPivot('is_active');
    }

    public function attendanceLogs()
    {
        return $this->hasMany(AttendanceLog::class);
    }
}

File: mobile.backup-20251230-070642/lib/models/shift.dart
class Shift {
  const Shift({
    required this.id,
    required this.name,
    required this.code,
    required this.startTime,
    required this.endTime,
    this.toleranceMinutes,
    required this.isDefault,
  });

  final int id;
  final String name;
  final String code;
  final String startTime;
  final String endTime;
  final int? toleranceMinutes;
  final bool isDefault;

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      code: json['code'] as String? ?? '',
      startTime: json['start_time']?.toString() ?? '',
      endTime: json['end_time']?.toString() ?? '',
      toleranceMinutes: json['tolerance_minutes'] as int?,
      isDefault: (json['is_default'] as bool?) ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'code': code,
      'start_time': startTime,
      'end_time': endTime,
      'tolerance_minutes': toleranceMinutes,
      'is_default': isDefault,
    };
  }
}
